api/controllers/UserController.js

```diff
@@ -398,36 +403,54 @@ module.exports = {
 
       const user = await User.create(request.payload);
       if (!user) {
-        // Delete sensitive fields before logging
-        delete request.payload.password;
-        delete request.payload.confirm_password;
-
         logger.warn(
           '[UserController->create] Create user failed',
-          { request: request.payload, fail: true },
+          {
+            request,
+            security: true,
+            fail: true,
+          },
         );
         throw Boom.badRequest();
       }
       logger.info(
-        `[UserController->create] User ${user._id.toString()} created successfully`,
+        `[UserController->create] User created successfully with ID ${user._id.toString()}`,
+        {
+          request,
+          security: true,
+          user: {
+            id: user._id.toString(),
```

Comment:
For the most part I just duplicated whatever variable was being used before. That's why it's different. I wanted to avoid refactoring stuff as much as possible while adding logs.

To your point: `user._id` property should really not be used whatsoever by us as humans. It's a [machine key of type `ObjectId`](https://docs.mongodb.com/manual/reference/method/ObjectId/) inherent to Mongo that can produce instability were we to shard/replicate our database (identical records would have different `ObjectId` values between DB replicas).

I know this and Serban knows it, but we haven't ever ticketed this bug which is built into the foundation of HID API. I will ticket it and come back here with a link. So the real bug is that we should rarely be referencing `_id` (except in a future situation where we've replicated the DB and want to log failures specific to one replica), and thus we'd never need to `toString()` a value which is already a plain string.

Ping @teodorescuserban to make sure I got the details right before ticketing. Please just gimme 👍 / 👎

Reply:
HID-2101